sealtk/core/AbstractProxyModel.cpp

```diff
@@ -32,9 +34,10 @@ bool AbstractProxyModel::isValidData(QVariant const& data, int role)
 {
   switch (role)
   {
-    // String comparisons
-    case core::NameRole:
     case core::UniqueIdentityRole:
+      return data.canConvert<QUuid>();
+
+    case core::NameRole:
       return data.canConvert<QString>();
```

Comment:
Hmm... integers can, of course, be converted to strings. Still, do you think this should be instead:
```c++
      switch (data.type())
      {
        case QVariant::Int:
        case QVariant::UInt:
        case QVariant::LongLong:
        case QVariant::ULongLong:
          return true;
        default:
          return data.canConvert<QString>();
      }
```
...?

Reply:
It's potentially less confusing to have both cases list out all the possibilities. Changed to above.